File: entsoe-client/src/main/java/org/example/model/common/MarketParticipantMRID.java
package org.example.model.common;

import javax.xml.bind.annotation.XmlAccessType;
import javax.xml.bind.annotation.XmlAccessorType;
import javax.xml.bind.annotation.XmlAttribute;
import javax.xml.bind.annotation.XmlValue;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

// Market Participant mRID class with coding scheme attribute
@XmlAccessorType(XmlAccessType.FIELD)
@Data
@NoArgsConstructor
@AllArgsConstructor
public class MarketParticipantMRID {

  @XmlAttribute(name = "codingScheme")
  private String codingScheme;

  @XmlValue private String value;
}

File: entsoe_client/src/entsoe_client/model/market/market_participant_mrid.py
"""Market participant MRID model for ENTSO-E market documents."""

from pydantic_xml import BaseXmlModel, attr


class MarketParticipantMRID(BaseXmlModel):  # Namespace-agnostic model
    """Market participant MRID for market documents."""

    value: str
    coding_scheme: str | None = attr(name="codingScheme")
